manim/renderer/cairo_renderer.py

```diff
@@ -0,0 +1,294 @@
+import numpy as np
+from .. import config, camera_config, file_writer_config, logger
+from ..utils.iterables import list_update
+from ..utils.exceptions import EndSceneEarlyException
+from ..utils.hashing import get_hash_from_play_call, get_hash_from_wait_call
+from ..constants import DEFAULT_WAIT_TIME
+from ..scene.scene_file_writer import SceneFileWriter
+
+
+def handle_caching_play(func):
```

Comment:
I moved this, but the skipping/caching/file writer handling is in need of some serious refactoring before we can move it into other renderers. It needs to be decoupled from skipping, use the same logic for wait and play (since most of the lines are exactly the same anyway), and probably shouldn't be a decorator.

Reply:
Yes, maybe we should do a decorator that will handle skipping flags and convert `handle_caching_play/wait` into a proper function that would be called at the beginning of `scene.play` and `wait? 
As now scene and renderer are two things, it would make way more sense. 

The issue is, handle_caching_play needs to be called before `handle_play_like_call`. So if a refactor is done, my opinion is that `handle_play_like_call` should be refactored and should incorporate all the caching stuff. As such, this decoratopr could handle skipping as well, and we won't have two decorators for a same function.

Reply:
Possibly, though scene skipping, scene caching, and handling the file writer in `handle_play_like_call` are all independent of each other. If they were handled separately it'd be easier to enable/disable them separately and to add new features later on.